src/engine/interop/specification.md

```diff
@@ -0,0 +1,148 @@
+
+## Core
+
+### engine_preparePayload
+
+#### Parameters
+1. `Object` - The payload attributes:
+- `parentHash`: `DATA`, 32 Bytes - hash of the parent block
+- `timestamp`: `QUANTITY`, 64 Bits - value for the `timestamp` field of the new payload
+- `random`: `DATA`, 32 Bytes - value for the `random` field of the new payload
+- `feeRecipient`: `DATA`, 20 Bytes - suggested value for the `coinbase` field of the new payload
+
+#### Returns
+1. `payloadId`: `QUANTITY`, 64 Bits - identifier of the payload building process or an error
+
+#### Specification
+
+1. Given provided field value set client software **SHOULD** build the initial version of the payload which has an empty transaction set.
+
+2. Client software **SHOULD** start the process of updating the payload. The strategy of this process is implementation dependent. The default strategy would be to keep the transaction set up-to-date with the state of local mempool.
+
+3. Client software **SHOULD** stop the updating process either by finishing to serve the [**`engine_getPayload`**](#engine_getPayload) call with the same `payloadId` value or when [`SECONDS_PER_SLOT`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#time-parameters-1) (currently set to 12 in the Mainnet configuration) seconds have passed since the point in time identified by the `timestamp` parameter.
+
+4. Client software **MUST** set the payload field values according to the set of parameters passed in the call to this method with exception for the `feeRecipient`. The `coinbase` field value **MAY** stem from what is specified by the `feeRecipient` parameter.
+
+5. Client software **SHOULD** respond with `2: Action not allowed` error if the sync process is in progress.
+
+6. Client software **MUST** respond with `4: Missing resource` error if the parent block is unknown.
+
+### engine_getPayload
+
+#### Parameters
+1. `payloadId`: `QUANTITY`, 64 Bits - Identifier of the payload building process
+
+#### Returns
+`Object|Error` - Either instance of [`ExecutionPayload`](#ExecutionPayload) or an error
+
+#### Specification
+
+1. Given the `payloadId` client software **MUST** respond with the most recent version of the payload that is available in the corresponding building process at the time of receiving the call.
+
+2. The call **MUST** be responded with `4: Missing resource` error if the building process identified by the `payloadId` doesn't exist.
+
+3. Client software **MAY** stop the corresponding building process after serving this call.
+
+### engine_executePayload
+
+#### Parameters
+1. `Object` - Instance of [`ExecutionPayload`](#ExecutionPayload)
+
+#### Returns
+`Object` - Response object:
+1. `status`: `String` - the result of the payload execution:
+  - `VALID` - given payload is valid
+  - `INVALID` - given payload is invalid
+  - `SYNCING` - sync process is in progress
```

Comment:
I think if it's syncing it should return an error?

Reply:
This is how sync process piggy backs on the core Engine API functions. And when `SYNCING` status turns with `VALID/INVALID` consensus client may conclude that the sync process has finished and node is ready to operate in a fully functional mode. Also, during the sync an execution client may store payloads for further usage and responding with error in this case wouldn't be correct.

Reply:
Ah okay got it!